### PR TITLE
chore: upgrade to npm 7 and superset-ui 0.17.9

### LINKED ISSUE #13067: [TODO] Upgrade to npm 7
https://github.blog/2021-02-02-npm-7-is-now-generally-available/

`npm 7` is now generally available and your local npm may have been upgraded if you have auto-update turned on.

But it has some breaking changes, most notably the new lockfile format and  how it handles `peerDependencies`. I did some testing and it seems the behavior for `npm link` has also changed---previously it will run `npm install` in linked packages, but now it doesn't (there is an open issue for this: https://github.com/npm/cli/issues/2339)

Extra care is needed to update `package-lock.json` and package dependencies in `superset-ui` may have to be updated to accommodate the new peer dependency behavior.

Every committer MUST be notified to upgrade their local `npm`, too, when we do upgrade to npm 7, because of these breaking changes.

cc @rusackas  @villebro @kgabryje @maloun96 @amitmiran137

### PR DESCRIPTION
**Pull request by ktmud**
### SUMMARY

Closes #13067 

#13079   upgraded `package-lock.json` to npm 7 format (`lockfileversion: 2`), which breaks the current workflow with `npm link` (although #13058 has downgraded it to `lockfileversion: 1` again):

1. Now you have to manually run `npm install` in each package you linked
2. A simple `npm install` will fail because of peer dependency resolution. You should use `npm install --legacy-peer-deps` until we resolve the peer dependency conflicts (seems solvable by upgrading to React 17). 

Instead of reverting, let's see what else we need to do to make the Superset codebase work with npm 7.

This PR updates the following:

1. Lock package.json [`engines`](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#engines) to npm 7
2. Upgrade Superset UI packages to the latest, including `@superset-ui/legacy-plugin-chart-map-box` which was left out of recent upgrades for a while... This brings in a couple of new commits in Superset UI:
    https://github.com/apache-superset/superset-ui/compare/v0.17.8...v0.17.9
    which adds a "sort by metric" checkbox for charts with single metric.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by villebro — APPROVED** [review]
LGTM and tested on node v12.20.2 + npm 7.5.4. It seems the `node:12` docker image is still on npm 6, as I'm getting this when running `docker-compose`: "npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!"

**Comment by ktmud** [comment]
@villebro Let me update the Dockerfile in this PR, too.

**Comment by ktmud** [comment]
Apparently #13058 has downgraded the lockfile to npm 6 format again.  To avoid future flip flops, I added some instructions in `CONTRIBUTING.md`.  @villebro could you take another look?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -42,6 +42,10 @@ little bit helps, and credit will always be given.
       - [Merging](#merging)
       - [Post-merge Responsibility](#post-merge-responsibility)
   - [Design Guidelines](#design-guidelines)
+    - [Capitalization guidelines](#capitalization-guidelines)
+      - [Sentence case](#sentence-case)
+      - [How to refer to UI elements](#how-to-refer-to-ui-elements)
+      - [\*\*Exceptions to sentence case:](#exceptions-to-sentence-case)
```
Comment (ktmud) [inline CONTRIBUTING.md:48]: Updates from Markdown formatter

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -464,7 +473,7 @@ Frontend assets (TypeScript, JavaScript, CSS, and images) must be compiled in or
 
 ##### nvm and node
 
-First, be sure you are using recent versions of NodeJS and npm. We recommend using [nvm](https://github.com/nvm-sh/nvm) to manage your node environment:
+First, be sure you are using recent versions of node.js and npm. We recommend using [nvm](https://github.com/nvm-sh/nvm) to manage your node environment:
```
Comment (villebro) [inline CONTRIBUTING.md:476]: I did some quick googling, and it appears the commonly used name is Node.js with a capital N: https://en.wikipedia.org/wiki/Node.js

**Review by villebro — APPROVED** [review]
LGTM, minor comment about Node.js name

**Comment by michael-s-molina** [comment]
@ktmud Thanks for upgrading to NPM 7! I didn't noticed that change in #13079.

**Pull request merged**